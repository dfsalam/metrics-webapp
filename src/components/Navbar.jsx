import React from 'react';
import { NavLink } from 'react-router-dom';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav>
    <NavLink to="/">
      <span className="material-symbols-outlined">
        arrow_back_ios_new
      </span>
    </NavLink>
    <h1 className="nav-title">Polution Data</h1>
    <div>
      <span className="material-symbols-outlined">
        keyboard_voice
      </span>
      <span className="material-symbols-outlined">
        settings
      </span>
    </div>

  </nav>
);

export default Navbar;
