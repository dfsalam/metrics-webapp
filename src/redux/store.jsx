import { configureStore } from '@reduxjs/toolkit';
import townsReducer from './towns/townsSlice';

const store = configureStore({
  reducer: {
    towns: townsReducer,
  },
});

export default store;
