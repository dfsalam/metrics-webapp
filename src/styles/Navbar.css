@import url('https://fonts.googleapis.com/css2?family=Lato:wght@400;700&display=swap');
@import url('https://fonts.cdnfonts.com/css/gill-sans');
@import url("https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@20..48,100..700,0..1,-50..200");

nav {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #ec4c8a;
}

.material-symbols-outlined {
  color: white;
  font-variation-settings: 'FILL' 1;
  padding: 0 5px;
}

.nav-title {
  color: white;
  font-family: 'Lato', serif;
  font-weight: 400;
  letter-spacing: 1px;
}
