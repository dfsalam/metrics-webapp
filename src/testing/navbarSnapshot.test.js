import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from '../components/Navbar';
import store from '../redux/store';

const provider = (component) => (
  <Provider store={store}>
    <BrowserRouter>{component}</BrowserRouter>
  </Provider>
);

describe('Components render correctly', () => {
  test('Navbar', () => {
    const tree = renderer
      .create(provider(<Navbar />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
