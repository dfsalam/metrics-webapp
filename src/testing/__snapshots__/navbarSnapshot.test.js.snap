// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render correctly Navbar 1`] = `
<nav>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <span
      className="material-symbols-outlined"
    >
      arrow_back_ios_new
    </span>
  </a>
  <h1
    className="nav-title"
  >
    Polution Data
  </h1>
  <div>
    <span
      className="material-symbols-outlined"
    >
      keyboard_voice
    </span>
    <span
      className="material-symbols-outlined"
    >
      settings
    </span>
  </div>
</nav>
`;
